### PR TITLE
Methylation qc docker

### PR DESCRIPTION
**Pull request by itamark-ug**

### CODE REVIEW

File: .github/workflows/build-methylationqc-docker-for-apl.yml [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+      - name: Extract build parameters
+        run: |
+          echo "BUILD_TIMESTAMP=$(date '+%F_%H:%M:%S')" >> ${GITHUB_ENV}
+          echo "BUILD_VERSION=$(git rev-parse --short HEAD)" >> ${GITHUB_ENV}
+
+      - name: Build and push methylation_qc docker image
+        uses: docker/build-push-action@v3
+        with:
+          context: .
+          file: Dockerfile.methylationqc
```
Comment (inbalzelig-ugi) [inline .github/workflows/build-methylationqc-docker-for-apl.yml]: Do you want to put the Dockerfile under setup/methylationqc as well?

File: .github/workflows/build-methylationqc-docker-for-apl.yml [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+          mask-aws-account-id: false
+
+      - uses: actions/checkout@v3
+        with:
+          submodules: true
+
+      # setup python and update the version in pyproject.toml
+      - uses: actions/setup-python@v4
+        with:
+          python-version: "3.11"
```
Comment (inbalzelig-ugi) [inline .github/workflows/build-methylationqc-docker-for-apl.yml]: In your pyproject.toml you use >=3.9,<3.10, and your docker based on python 3.9. 
Just make sure everything is aligned

File: setup/methylationqc/data/meth_hg38-chr20_Lambda_pUC19.interval_list [hunk 3]
```diff
@@ -0,0 +1,461 @@
+@HD	VN:1.6	SO:coordinate
+@SQ	SN:chr1	LN:248956422	M5:2648ae1bacce4ec4b6cf337dcae37816	UR:file:///data/hg38_Lambda_pUC19.fa
```
Comment (inbalzelig-ugi) [inline setup/methylationqc/data/meth_hg38-chr20_Lambda_pUC19.interval_list:2]: @GatKrieger, maybe we should remove all @SQ (or keep just the ones relevant for the intervals in this list?)

File: Dockerfile.methylationqc [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+COPY --from=build /app/bin /app/bin
+COPY --from=build /app/gatk/gatk-4.5.0.0 /app/gatk
+COPY . .
+COPY ./setup/methylationqc/pyproject.toml ./pyproject.toml
+COPY ./setup/methylationqc/data/meth_hg38-chr20_Lambda_pUC19.interval_list /app/interval_list/meth_hg38-chr20_Lambda_pUC19.interval_list
+
+ENV PATH="/app/gatk:$PATH"
+ENV PATH="/app/bin:$PATH"
+
+WORKDIR /app/VariantCalling
```
Comment (inbalzelig-ugi) [inline Dockerfile.methylationqc:96]: You have multiple "WORKDIR /app/VariantCalling" lines, make sure all are needed

File: Dockerfile.methylationqc [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+
+# most of the time we would like to run python scripts (which will come in the command)
+# ENTRYPOINT [ "/opt/venv/bin/python" ]
```
Comment (inbalzelig-ugi) [inline Dockerfile.methylationqc:99]: Remove the comment

**Review by inbalzelig-ugi — APPROVED** [review]

**Review by inbalzelig-ugi — APPROVED** [review]

File: .github/workflows/build-methylationqc-docker-for-apl.yml (hunk 1 above)
Comment (itamark-ug) [reply .github/workflows/build-methylationqc-docker-for-apl.yml]: thought about it :-), wanted to first pass the integration than handle this stuff

File: .github/workflows/build-methylationqc-docker-for-apl.yml (hunk 2 above)
Comment (itamark-ug) [reply .github/workflows/build-methylationqc-docker-for-apl.yml]: it is not that critical, this is just for the poetry installation, not effecting the docker (which is based on python 3.9)

**Pull request merged**